src/Api/Serializer/BasicDiscussionSerializer.php

```diff
@@ -35,7 +46,7 @@ protected function getDefaultAttributes($discussion)
 
         return [
             'title' => $discussion->title,
-            'slug'  => $discussion->slug,
+            'slug' =>  $this->slugManager->forResource(Discussion::class)->toSlug($discussion),
```

Comment:
Wouldn't it be easier to have a method on the abstract model instead?

```php
$discussion->slugger->slug();
```

So that we resolve the slug manager late here as well, plus we can reduce the overhead on every, single class that tries to resolve the slug manager..

Reply:
I'm a bit confused how that would work tbh. I'm very wary about adding stuff to AbstractModel unless it absolutely affects ALL  models, which isn't the case here.

Reply:
What if it's a trait one can apply to the model, eg Discussion has the Sluggable trait. It adds that trait, sets the property `protected $slugger = DiscussionSlugger::class;` and the trait handles the rest?

Reply:
I don't like this. It adds unnecessary static stuff to the models, and I'd really prefer to move away from that as much as we possibly can because it's unnecessary leakage of global state. It also makes the extension mechanism a lot more hacky (similar to what I had to do with #2460), which I really didn't like and will revisit again after stable.